guardian/elastic-search-monitor
# Update logging approach to reinstate application logging

## What does this change?

Application logging has not been working for this Lambda for several months. I believe the problem was introduced by upgrading to v2 of `slf4j-simple` in https://github.com/guardian/elastic-search-monitor/pull/44 (after downgrading to v1 I can see logs coming through again). 

I had a quick look into this, but I don't completely understand why the upgrade broke logging within the Lambda runtime. In order to get the logging working again quickly I have migrated to [another approach that AWS recommends](https://docs.aws.amazon.com/lambda/latest/dg/java-logging.html#java-logging-advanced), which avoids the need for managing extra third-party dependencies.

In order to achieve this I've also upgraded GuCDK as this now sets up JSON logging by default: https://github.com/guardian/cdk/pull/2260.

## How to test

I've deployed this to confirm that application logs start coming through to ELK again.

## How can we measure success?

* It's easier to debug this application
* We have fewer third-party dependencies to keep up to date

## Have we considered potential risks?

I don't think there are any particular risks associated with this PR.
